Add keyboard controls

sharkwouter opened the pull request:
Being able to control the game with the keyboard as well is a bit more comfortable and it makes easier to port it to different systems. I want to try to port this game to the Playstation Vita later. For now this adds the following:

- Both moving the mouse and the arrow keys can move the selector square on the board.
- When the mouse is not over the board, the keyboard can be used.
- The spacebar now performs the same action as the mouse button.
- When moving out of the bounds of the board with the keyboard, the selector will warp to the opposite side of it.

What needs to be changed is that using the keyboard should deactivate the mouse until it is moved again and some additional functions are needed to get rid of some duplicate code. That I'm still working on.

JoseTomasTocino commented:
Good job @sharkwouter ! Looking forward to the finished pull request, looking good!

sharkwouter commented:
Okay, this should be good now. I've added the following the the last few commits:

- If the keyboard has been used to move the selector square, it will no longer be moved my mouse movement until the user clicks on a gem.
- I've removed some duplicate code.
- The default starting position of the selector square was changed from 0,0 to 3,3.

Please let me know if you find any issues with testing.
